### ISSUE TITLE
Make fish swim away from players

### ISSUE THREAD

**Issue by Aceplante** [CartoonishVillain/VillainousFishAI#1]
As in quickly. Whenever a player in water is near them, they quickly attempt to dart away, forcing the player to either shoot them with a trident/bow, or do so from outside the water/riverbank

**Comment by Keaton42** [comment]
I agree. It's too easy to hunt fish by hand because of how dumb they are! I want fishing rod fishing to be more viable

**Comment by CartoonishVillain** [comment]
The reason I didn't do this, is that this is a feature already in vanilla minecraft. They're just very bad at it!

**Comment by Aceplante** [comment]
Could you perhaps make them have a faster reaction time then?

**Comment by CartoonishVillain** [comment]
That maybe I can do. I'll need to look at it some more. Would probably need to balance the goals that fish have. 

One of the biggest things that cause them to not flee quickly is that fish typically spawn in or near schools of fish and they will dramatically slow down to swim with a school of fish. To that end even when a whole school of fish is trying to swim away at once, they're all slowing down to ensure each member of the school sticks around. 

I make no guarantees that I can make this work as I'd rather preserve the school of fish behavior rather than make them swim faster, and if I can't figure out a good solution for both I'll defer to vanilla behavior.

**Comment by Aceplante** [comment]
Maybe when attacked or one dies, they no longer care and flee randomly with no regard for other fish. They flee in a random direction and ignore normal school behavior. They can regroup later.

**Comment by CartoonishVillain** [comment]
Changelog for version 1.2:
* Version bump to 1.21.1
* Fish will now more vigorously swim away from all sources
* I've also replaced the vanilla avoid goal for players to make them more vigorously swim away from players too
* They can only move so fast, even with my efforts to continuously pour metaphorical gas on the engine.
-----
It seems now that the school of fish behavior can still get the fish stuck, but overall doesn't hinder their speed until then now

**Comment by Keaton42** [comment]
Wow thank you!! You've been doing great work!

**Comment by martinmeza0** [comment]
Hi, its possible to backport this to 1.20.1? if you can't. Can i have your permission to backport it and send you the code?
